packages/babel-core/src/transformation/file/merge-map.ts

```diff
@@ -1,321 +1,25 @@
   return result;
 }
 
-function makeMappingKey(item: { line: number; columnStart: number }) {
-  return `${item.line}/${item.columnStart}`;
-}
-
-function eachOverlappingGeneratedOutputRange(
-  outputFile: ResolvedFileMappings,
-  inputGeneratedRange: ResolvedGeneratedRange,
-  callback: (range: ResolvedGeneratedRange) => unknown,
-) {
-  // Find the Babel-generated mappings that overlap with this range in the
-  // input sourcemap. Generated locations within the input sourcemap
-  // correspond with the original locations in the map Babel generates.
-  const overlappingOriginal = filterApplicableOriginalRanges(
-    outputFile,
-    inputGeneratedRange,
-  );
-
-  for (const { generated } of overlappingOriginal) {
-    for (const item of generated) {
-      callback(item);
-    }
-  }
-}
-
-function filterApplicableOriginalRanges(
-  { mappings }: ResolvedFileMappings,
-  { line, columnStart, columnEnd }: ResolvedGeneratedRange,
-): OriginalMappings {
-  // The mapping array is sorted by original location, so we can
-  // binary-search it for the overlapping ranges.
-  return filterSortedArray(mappings, ({ original: outOriginal }) => {
-    if (line > outOriginal.line) return -1;
-    if (line < outOriginal.line) return 1;
-
-    if (columnStart >= outOriginal.columnEnd) return -1;
-    if (columnEnd <= outOriginal.columnStart) return 1;
-
-    return 0;
-  });
-}
-
-function eachInputGeneratedRange(
-  map: ResolvedMappings,
-  callback: (
-    c: ResolvedGeneratedRange,
-    b: ResolvedOriginalRange,
-    a: ResolvedSource,
-  ) => unknown,
-) {
-  for (const { source, mappings } of map.sources) {
-    for (const { original, generated } of mappings) {
-      for (const item of generated) {
-        callback(item, original, source);
-      }
-    }
-  }
-}
-
-type ResolvedMappings = {
-  file: string | undefined | null;
-  sourceRoot: string | undefined | null;
-  sources: Array<ResolvedFileMappings>;
-};
-
-type ResolvedFileMappings = {
-  source: ResolvedSource;
-  mappings: OriginalMappings;
-};
-
-type OriginalMappings = Array<{
-  original: ResolvedOriginalRange;
-  generated: Array<ResolvedGeneratedRange>;
-}>;
-
-type ResolvedSource = {
-  path: string;
-  content: string | null;
-};
-
-type ResolvedOriginalRange = {
-  line: number;
-  columnStart: number;
-  columnEnd: number;
-  name: string | null;
-};
-
-type ResolvedGeneratedRange = {
-  line: number;
-  columnStart: number;
-  columnEnd: number;
-};
-
-function buildMappingData(map: SourceMap): ResolvedMappings {
-  const consumer = new sourceMap.SourceMapConsumer({
+function rootless(map: SourceMap): SourceMap {
+  return {
     ...map,
 
-    // This is a bit hack. .addMapping expects source values to be relative,
-    // but eachMapping returns mappings with absolute paths. To avoid that
-    // incompatibility, we leave the sourceRoot out here and add it to the
-    // final map at the end instead.
+    // This is a bit hack. Remapping will create absolute sources in our
```

Comment:
Is it true for `@ampproject/remapping` too? In the readme, the `helloworld.js` doesn't seem to be transformed to absolute.

Reply:
Unfortunately so: https://github.com/ampproject/remapping/blob/fcc9f442f43249389dc946cbe8bb3736a8033d6a/src/source-map-tree.ts#L115.

> In the readme, the helloworld.js doesn't seem to be transformed to absolute.

The readme example doesn't have a `sourceRoot`. If one were included, say `src`, then the output would partially match `{ sourceRoot: undefined, sources: ['src/helloworld.js'] }`.